extension/js/common/core/msg-block-parser.ts

```diff
@@ -84,20 +85,29 @@ export class MsgBlockParser {
     return { blocks, subject: decoded.subject, isRichText, webReplyToken };
   };
 
-  public static extractFcAttachments = (decryptedContent: string, blocks: MsgBlock[]) => {
+  public static extractFcAttachments = async (decryptedContent: string, blocks: MsgBlock[]) => {
     // these tags were created by FlowCrypt exclusively, so the structure is rigid (not arbitrary html)
     // `<a href="${attachment.url}" class="cryptup_file" cryptup-data="${fcData}">${linkText}</a>\n`
     // thus we use RegEx so that it works on both browser and node
+    let error = '';
+    const fcAttachmentsPattern = /<a\s+href="([^"]+)"\s+class="cryptup_file"\s+cryptup-data="([^"]+)"\s*>[^<]+<\/a>\n?/gm;
```

Comment:
Thank you for the demo, @martgil.
My initial suggestion was to include the URL in the regex, like
`/<a\s+href="(https?:\/\/flowcrypt\.s3\.amazonaws\.com\/[^"]+)"\s+class="cryptup_file"\s+cryptup-data="([^"]+)"\s*>[^<]+<\/a>\n?/gm`
so any external URL will be naturally skipped and result in simple text rendering like
![image](https://user-images.githubusercontent.com/23335944/204825010-6641ad0e-f3a1-4c5b-9b2d-d33d17cae121.png)
Your implementation gives user some insight into a possible attack, but vaguely.
![image](https://user-images.githubusercontent.com/23335944/204835476-c0c8fdc1-c97b-4466-b35e-e82e54ed1032.png)
We need to decide first whether we actually need this extra UI (and making the user press 'OK' may also be redundant), especially by complicating the code.
If we do need to display something special for this case, we can think about simplifying the parsing/rendering code next.
@ioanmo226, @tomholub What would you vote for?

Reply:
Well, I think we need to show original content (like @rrrooommmaaa suggested) + notify user via `toast` (something like `Skipping attachment rendering and show original content because attachment url is modified/invalid`

Reply:
Whatever is simpler to implement. Could be:
 - toast
 - modal
 - changing the content of the message to include `[skipped attachment due to invalid url]`